e2e/nomostest/nt.go

```diff
@@ -596,6 +593,40 @@ func (nt *NT) newPortForwarder(ns, deployment, port string, opts ...portforwarde
 	)
 }
 
+func (nt *NT) startPortForwarder(ns, deployment string, pf *portforwarder.PortForwarder) {
+	ctx, cancel := context.WithCancel(nt.Context)
+	eventCh := pf.Start(ctx)
+	doneCh := make(chan struct{})
+	readyCh := make(chan struct{})
+	nt.T.Cleanup(func() {
+		nt.T.Logf("killing port-forward %s/%s", ns, deployment)
+		cancel()
+		<-doneCh
+		nt.T.Logf("finished waiting for port-forward %s/%s to exit", ns, deployment)
+	})
+	go func() {
+		defer close(doneCh)
+		// first event should be Init. signal if it's ready
+		event := <-eventCh
+		if event.Type == portforwarder.InitEvent {
```

Comment:
nit: Minor simplification: You could just do this in the for loop with a switch. It would panic on close if you received init twice.